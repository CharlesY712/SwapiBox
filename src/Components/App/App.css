.App {
  text-align: center;
}

.people.active {
  color: magenta;
  background-color: salmon;
}