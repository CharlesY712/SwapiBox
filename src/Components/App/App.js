import React, { Component } from "react";
import { Route, NavLink, Link } from "react-router-dom";
import "./App.css";
import CardContainer from "../CardContainer/CardContainer";
import Welcome from "../Welcome/Welcome";
import fetchOpeningCrawl from "../API/fetchOpeningCrawl";
import fetchPeople from "../API/fetchPeople";

class App extends Component {
  constructor() {
    super();
    this.state = {
      crawl: null,
      people: [{}, {}],
      planets: [],
      vehicles: [],
      favorites: []
    };
  }

  async componentDidMount() {
    const crawl = await fetchOpeningCrawl();
    this.setState({ crawl });
    const peopleObj = await fetchPeople();
    const people = peopleObj.results;
    this.setState({ people });
    console.log(this.state.people);
  }

  componentDidUpdate() {}

  render() {
    return (
      <div className="App">
        <Link to="/" className='swapiLink'>
          <h1 className="App-title">SwapiBox</h1>
        </Link>
        <NavLink to="/people" className="people">
          People
        </NavLink>
        <NavLink to="/vehicles" className="vehicles">Vehicles</NavLink>
        <NavLink to="/planets" className="planets">Planets</NavLink>
        <NavLink to="/favorites" className="favorites">0 Favorites</NavLink>
        {this.state.crawl && (
          <div>
            <Route
              exact path="/"
              render={() => <Welcome crawl={this.state.crawl} />}
            />
            <Route
              path="/people"
              render={() => <CardContainer people={this.state.people} />}
            />
            <Route
              path="/planets"
              render={() => <CardContainer planets={this.state.planets} />}
            />
            <Route
              path="/vehicles"
              render={() => <CardContainer vehicles={this.state.vehicles} />}
            />
            <Route
              path="/favorites"
              render={() => <CardContainer favorites={this.state.favorites} />}
            />
          </div>
        )}
      </div>
    );
  }
}

export default App;
