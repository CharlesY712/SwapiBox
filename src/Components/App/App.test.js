import React from 'react';
import { shallow } from 'enzyme';
import App from './App';


describe('App', () =>{

  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<App />);
    /* eslint-disable no-undef */
    window.fetch = jest.fn().mockImplementation(() => Promise.resolve({
      json: () => Promise.resolve('banana')
    }));
    /* eslint-enable no-undef */
  });

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should fetch opening crawl data on mount', async () => {
    // console.log(wrapper.state())
    // console.log(wrapper.instance().componentDidMount();
  });

  it('should set the state on mount', () => {

  });
});