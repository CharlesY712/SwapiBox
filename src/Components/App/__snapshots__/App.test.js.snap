// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <header
          className="header"
>
          <Link
                    className="swapiLink"
                    replace={false}
                    to="/"
          >
                    <h1
                              className="App-title"
                    >
                              SwapiBox
                    </h1>
          </Link>
          <NavLink
                    activeClassName="active"
                    ariaCurrent="true"
                    className="people"
                    to="/people"
          >
                    People
          </NavLink>
          <NavLink
                    activeClassName="active"
                    ariaCurrent="true"
                    className="planets"
                    to="/planets"
          >
                    Planets
          </NavLink>
          <NavLink
                    activeClassName="active"
                    ariaCurrent="true"
                    className="vehicles"
                    to="/vehicles"
          >
                    Vehicles
          </NavLink>
          <NavLink
                    activeClassName="active"
                    ariaCurrent="true"
                    className="favorites"
                    to="/favorites"
          >
                    0 Favorites
          </NavLink>
</header>,
        null,
      ],
      "className": "App",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <Link
              className="swapiLink"
              replace={false}
              to="/"
>
              <h1
                            className="App-title"
              >
                            SwapiBox
              </h1>
</Link>,
            <NavLink
              activeClassName="active"
              ariaCurrent="true"
              className="people"
              to="/people"
>
              People
</NavLink>,
            <NavLink
              activeClassName="active"
              ariaCurrent="true"
              className="planets"
              to="/planets"
>
              Planets
</NavLink>,
            <NavLink
              activeClassName="active"
              ariaCurrent="true"
              className="vehicles"
              to="/vehicles"
>
              Vehicles
</NavLink>,
            <NavLink
              activeClassName="active"
              ariaCurrent="true"
              className="favorites"
              to="/favorites"
>
              0 Favorites
</NavLink>,
          ],
          "className": "header",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <h1
                className="App-title"
>
                SwapiBox
</h1>,
              "className": "swapiLink",
              "replace": false,
              "to": "/",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "SwapiBox",
                "className": "App-title",
              },
              "ref": null,
              "rendered": "SwapiBox",
              "type": "h1",
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "activeClassName": "active",
              "ariaCurrent": "true",
              "children": "People",
              "className": "people",
              "to": "/people",
            },
            "ref": null,
            "rendered": "People",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "activeClassName": "active",
              "ariaCurrent": "true",
              "children": "Planets",
              "className": "planets",
              "to": "/planets",
            },
            "ref": null,
            "rendered": "Planets",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "activeClassName": "active",
              "ariaCurrent": "true",
              "children": "Vehicles",
              "className": "vehicles",
              "to": "/vehicles",
            },
            "ref": null,
            "rendered": "Vehicles",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "activeClassName": "active",
              "ariaCurrent": "true",
              "children": "0 Favorites",
              "className": "favorites",
              "to": "/favorites",
            },
            "ref": null,
            "rendered": "0 Favorites",
            "type": [Function],
          },
        ],
        "type": "header",
      },
      null,
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <header
            className="header"
>
            <Link
                        className="swapiLink"
                        replace={false}
                        to="/"
            >
                        <h1
                                    className="App-title"
                        >
                                    SwapiBox
                        </h1>
            </Link>
            <NavLink
                        activeClassName="active"
                        ariaCurrent="true"
                        className="people"
                        to="/people"
            >
                        People
            </NavLink>
            <NavLink
                        activeClassName="active"
                        ariaCurrent="true"
                        className="planets"
                        to="/planets"
            >
                        Planets
            </NavLink>
            <NavLink
                        activeClassName="active"
                        ariaCurrent="true"
                        className="vehicles"
                        to="/vehicles"
            >
                        Vehicles
            </NavLink>
            <NavLink
                        activeClassName="active"
                        ariaCurrent="true"
                        className="favorites"
                        to="/favorites"
            >
                        0 Favorites
            </NavLink>
</header>,
          null,
        ],
        "className": "App",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <Link
                className="swapiLink"
                replace={false}
                to="/"
>
                <h1
                                className="App-title"
                >
                                SwapiBox
                </h1>
</Link>,
              <NavLink
                activeClassName="active"
                ariaCurrent="true"
                className="people"
                to="/people"
>
                People
</NavLink>,
              <NavLink
                activeClassName="active"
                ariaCurrent="true"
                className="planets"
                to="/planets"
>
                Planets
</NavLink>,
              <NavLink
                activeClassName="active"
                ariaCurrent="true"
                className="vehicles"
                to="/vehicles"
>
                Vehicles
</NavLink>,
              <NavLink
                activeClassName="active"
                ariaCurrent="true"
                className="favorites"
                to="/favorites"
>
                0 Favorites
</NavLink>,
            ],
            "className": "header",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": <h1
                  className="App-title"
>
                  SwapiBox
</h1>,
                "className": "swapiLink",
                "replace": false,
                "to": "/",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "SwapiBox",
                  "className": "App-title",
                },
                "ref": null,
                "rendered": "SwapiBox",
                "type": "h1",
              },
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "activeClassName": "active",
                "ariaCurrent": "true",
                "children": "People",
                "className": "people",
                "to": "/people",
              },
              "ref": null,
              "rendered": "People",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "activeClassName": "active",
                "ariaCurrent": "true",
                "children": "Planets",
                "className": "planets",
                "to": "/planets",
              },
              "ref": null,
              "rendered": "Planets",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "activeClassName": "active",
                "ariaCurrent": "true",
                "children": "Vehicles",
                "className": "vehicles",
                "to": "/vehicles",
              },
              "ref": null,
              "rendered": "Vehicles",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "activeClassName": "active",
                "ariaCurrent": "true",
                "children": "0 Favorites",
                "className": "favorites",
                "to": "/favorites",
              },
              "ref": null,
              "rendered": "0 Favorites",
              "type": [Function],
            },
          ],
          "type": "header",
        },
        null,
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
